src/main/java/com/questhelper/requirements/ItemSlots.java

```diff
@@ -43,21 +50,50 @@
 
 	private final int slotIdx;
 	private final String name;
+	private final InventorySlots inventorySlots;
 
 	ItemSlots(int slotIdx, String name)
 	{
 		this.slotIdx = slotIdx;
 		this.name = name;
+		this.inventorySlots = null;
 	}
-
-	public int getSlotIdx()
+	ItemSlots(int slotIdx, String name, InventorySlots slots)
 	{
-		return slotIdx;
+		this.slotIdx = slotIdx;
+		this.name = name;
+		this.inventorySlots = slots;
 	}
 
-	public String getName()
+	/**
+	 * Checks that a given item slot is empty.
+	 * If checking all equipment, inventory, or both, the predicate will be used to
+	 *
+	 * @param client the {@link Client} to check
+	 * @param predicate the predicate to use
+	 * @return true if ALL the items match the predicate via {@link Stream#allMatch(Predicate)}
+	 */
+	public boolean checkInventory(Client client, Predicate<Item> predicate)
 	{
-		return name;
+		// if we're checking all the equipment slots, inventory slots, or both
+		if (getInventorySlots() != null)
+		{
+			return getInventorySlots().checkInventory(client, predicate);
+		}
+		// otherwise check a specific slot
+		if (client.getLocalPlayer() == null || client.getLocalPlayer().getPlayerComposition() == null)
+		{
+			return true;
+		}
+
+		int[] equipment = client.getLocalPlayer().getPlayerComposition().getEquipmentIds();
+
+		if (equipment == null)
+		{
+			return true;
+		}
+
+		return getSlotIdx() >= 0 && equipment[getSlotIdx()] <= 512;
```

Comment:
the `getSlotIdx() >= 0` check doesn't *technically* need to be there since the `getInventorySlots() != null` check is there, but better safe than sorry